src/Compilers/CSharp/Portable/Symbols/AbstractTypeMap.cs

```diff
@@ -138,21 +138,29 @@ internal TypeSymbolWithAnnotations SubstituteType(TypeSymbolWithAnnotations prev
         /// In particular, if substitution makes type tuple compatible, transform it into a tuple type.
         /// </summary>
         internal TypeSymbolWithAnnotations SubstituteTypeWithTupleUnification(TypeSymbol previous)
```

Comment:
More general question: why do we have this API, instead of just using `SubstituteType`?

Reply:
Because the compiler does not match the language in the way that tuple types are handled. See https://github.com/dotnet/roslyn/issues/20648 which describes a set of changes that are planned to bring them into alignment.